xopt/generator.py

```diff
@@ -115,3 +115,21 @@ def model_dump(self, *args, **kwargs) -> dict[str, Any]:
         res.pop("supports_multi_objective", None)
 
         return res
+
+
+class StateOwner:
```

Comment:
Instead of adding a set_data method, can we just assign the attribute via `generator.data = data` we could probably do that for all generators in the base.py file instead of calling a method